api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -108,16 +109,33 @@ public void validate(Object obj, Errors errors) {
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
 			validateUnitsAreAmongAllowedConcepts(errors, order);
-			validateForRequireDrug(errors);
+			validateForRequireDrug(errors, order);
 		}
 	}
-	
-	private void validateForRequireDrug(Errors errors) {
+
+	private void validateForRequireDrug(Errors errors, DrugOrder order) {
 		//Reject if global property is set to specify a formulation for drug order
 		boolean requireDrug = Context.getAdministrationService().getGlobalPropertyValue(
-		    OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
-		if (requireDrug) {
-			ValidationUtils.rejectIfEmpty(errors, "drug", "DrugOrder.error.drugIsRequired");
+				OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
+		OrderService orderService = Context.getOrderService();
+
+
```

Comment:
What is in the ticket's acceptance criteria is what i think you should implement, the concept needs to be DRUG OTHER or of class drug, we have a design call in about 35min, i think we can confirm this with the others